File: sion_fasion_shop/ProyectoMVC/SionFashioWebApplicationMVC/SionFashioWebApplicationMVC/Models/factura.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace SionFashioWebApplicationMVC.Models;

public partial class factura
{
    [Key]
    public int id_factura { get; set; }

    [Required(ErrorMessage = "La fecha de emisión es obligatoria.")]
    [Column(TypeName = "datetime")]
    public DateTime fecha_emision_factura { get; set; }

    [Required(ErrorMessage = "El subtotal es obligatorio.")]
    [Range(0, 9999999999.99, ErrorMessage = "El subtotal debe estar entre 0 y 9999999999.99.")]
    [Column(TypeName = "decimal(10, 2)")]
    public decimal sub_total_factura { get; set; }

    [Required(ErrorMessage = "El impuesto es obligatorio.")]
    [Range(0, 9999999999.99, ErrorMessage = "El impuesto debe estar entre 0 y 9999999999.99.")]
    [Column(TypeName = "decimal(10, 2)")]
    public decimal impuesto_factura { get; set; }

    [Required(ErrorMessage = "El total es obligatorio.")]
    [Range(0, 9999999999.99, ErrorMessage = "El total debe estar entre 0 y 9999999999.99.")]
    [Column(TypeName = "decimal(10, 2)")]
    public decimal total_factura { get; set; }

    [Required(ErrorMessage = "El cliente es obligatorio.")]
    public int id_cliente { get; set; }

    [ForeignKey("id_cliente")]
    [InverseProperty("facturas")]
    public virtual cliente id_clienteNavigation { get; set; } = null!;

    [InverseProperty("id_facturaNavigation")]
    public virtual ICollection<metodos_de_pago> metodos_de_pagos { get; set; } = new List<metodos_de_pago>();

    [InverseProperty("id_facturaNavigation")]
    public virtual ICollection<ordenes_de_compra> ordenes_de_compras { get; set; } = new List<ordenes_de_compra>();

    [InverseProperty("id_facturaNavigation")]
    public virtual ICollection<producto> productos { get; set; } = new List<producto>();
}

File: sion_fasion_shop/Entregables/Trimestre V/Sistema de inventario en Node.Js/Bck-End SionFashion-Shop/models/factura.js
const mongoose = require('mongoose');
const { Schema } = mongoose;

const facturaSchema = new Schema({

    fecha_emision_factura: {
        type: Date,
        required: [true, 'La fecha de emisión es obligatoria.']
    },
    sub_total_factura: {
        type: Schema.Types.Decimal128,
        required: [true, 'El subtotal es obligatorio.'],
        min: [0, 'El subtotal debe estar entre 0 y 9999999999.99.'],
        max: [9999999999.99, 'El subtotal debe estar entre 0 y 9999999999.99.']
    },
    impuesto_factura: {
        type: Schema.Types.Decimal128,
        required: [true, 'El impuesto es obligatorio.'],
        min: [0, 'El impuesto debe estar entre 0 y 9999999999.99.'],
        max: [9999999999.99, 'El impuesto debe estar entre 0 y 9999999999.99.']
    },
    total_factura: {
        type: Schema.Types.Decimal128,
        required: [true, 'El total es obligatorio.'],
        min: [0, 'El total debe estar entre 0 y 9999999999.99.'],
        max: [9999999999.99, 'El total debe estar entre 0 y 9999999999.99.']
    },

    id_clienteNavigation: {
        type: Schema.Types.ObjectId,
        ref: 'Cliente' // Asegúrate de que este sea el nombre correcto del modelo cliente
    },
    metodos_de_pagos: [{
        type: Schema.Types.ObjectId,
        ref: 'MetodosDePago' // Asegúrate de que este sea el nombre correcto del modelo métodos de pago
    }],
    ordenes_de_compras: [{
        type: Schema.Types.ObjectId,
        ref: 'OrdenesDeCompra' // Asegúrate de que este sea el nombre correcto del modelo órdenes de compra
    }],
    productos: [{
        type: Schema.Types.ObjectId,
        ref: 'Producto' // Asegúrate de que este sea el nombre correcto del modelo producto
    }]
});

// Exportar el modelo
module.exports = mongoose.model('Factura', facturaSchema);
